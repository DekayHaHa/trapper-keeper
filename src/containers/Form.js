import React, { Component } from 'react';
import { addItem, addTitle } from '../actions';
import { connect } from 'react-redux';

export class Form extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      item: '',
      itemsList: []
    }
  }

  handleChange = (e) => {
    const { value, name } = e.target;
    this.setState({ [name]: value });
  }

  addIdea = () => {
    const { item, itemsList } = this.state
    this.setState({
      itemsList: [...itemsList, item],
      item: ''
    })
    // this.setState({ items: [...items, value] })  
  }

  handleSubmit = (e) => {
    e.preventDefault();

    // add card
  }

  renderItems = () => this.state.itemsList.map(item => <p>{item}</p>)

  render() {
    const { title, item, itemsList } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <input 
          type='text' 
          placeholder='Title...' 
          value={title}
          name='title'
          onChange={this.handleChange}
        />
        {itemsList.length > 0 && 
          this.renderItems()
        }
        <button onClick={this.addIdea}>+</button><input 
          type='text' 
          placeholder='Item...' 
          value={item}
          name='item'
          onChange={this.handleChange}
        />
        <button>Add Item</button>
        {/* add condition for global items array, if yes, renderItems */}
      </form>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addItem: (items) => dispatch(addItem(items)),
  addTitle: (title) => dispatch(addTitle(title))
});

export const mapStateToProps = (state) => ({
  items: state.items
});

export default connect(mapStateToProps, mapDispatchToProps)(Form);
