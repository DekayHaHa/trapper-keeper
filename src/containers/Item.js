// ignore thus file for now

// import React, { Component } from 'react';

// export class Item extends Component {
// 	constructor() {
// 		super();
// 		this.state = {
// 			text: this.props.item || '',
// 			isComplete: this.props.isComplete || false,
// 			id: this.props.id || null
// 		};
// 	}

// 	handleChange = e => {
// 		const { value, name } = e.target;
// 		const { text } = this.state
// 		this.setState({ [name]: value });
// 		if (text.length === 1) {
// 			this.props.addListItem(this.state)
// 		}

// 	}

// 	// sendListItem = () => {
// 	// 	this.props.addListItem(this.state.item)
// 	// 	this.setState({ item: '' })
// 	// }

// 	render() {
// 		const itemInput = <input type='text' name='item' value={this.state.text} onChange={this.handleChange} placeholder='item...' />
// 		return (
// 			<div>
// 				{itemInput}
// 				{/* <input type='text' name='item' value={this.state.item} onChange={this.handleChange} placeholder='item...' /> */}
// 			</div>
// 		);
// 	}
// }