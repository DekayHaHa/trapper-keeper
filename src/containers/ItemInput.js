import React, { Component } from 'react';

export class ItemInput extends Component {
	constructor(props) {
		super(props);
		this.state = {
			text: '',
			isComplete: false,
			id: 0
		};
	}

	componentWillMount() {
		const { text, isComplete, id } = this.props
		if (this.props.text) {
			console.log(this.props)
			this.setState({ text, isComplete, id })
		}
	}

	handleChange = e => {
		const { value, name } = e.target;
		this.setState({ [name]: value });
	}

	sendListItem = () => {
		const { addListItem, updateItem } = this.props
		if (addListItem) {
			addListItem(this.state.text)
			this.setState({ text: '' })
		} else {
			updateItem(this.state)
		}
	}

	render() {
		return (
			<div>
				<input type='text' name='text' value={this.state.text} onBlur={this.sendListItem} onChange={this.handleChange} placeholder='item...' />
			</div>
		);
	}
}