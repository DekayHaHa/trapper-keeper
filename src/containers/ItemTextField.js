import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TextField } from '@material-ui/core';

export class ItemTextField extends Component {
    constructor(props) {
        super(props);
        this.state = {
            text: '',
            isComplete: false,
            id: 0
        };
    }

    componentWillMount() {
        const { text, isComplete, id } = this.props;
        if (this.props.text) {
            this.setState({ text, isComplete, id });
        }
    }

    handleChange = e => {
        const { value, name } = e.target;
        this.setState({ [name]: value });
    }

    sendListItem = () => {
        console.log('in send list')
        const { addListItem, updateItem } = this.props;
        if (addListItem) {
            addListItem(this.state.text);
            this.setState({ text: '' });
        } else {
            updateItem(this.state);
        }
    }

    catchKey = (e) => {
        if(e.keyCode === 13) this.sendListItem();
    }

    render() {
        const { text } = this.state;
        return (
            <div>
                <TextField margin="dense" id="item" label="Item" type="text" name='text' value={text} onBlur={this.sendListItem}onChange={this.handleChange} onKeyDown={this.catchKey} fullWidth />
            </div>
        );
    }
}
ItemTextField.propTypes = {
    id: PropTypes.number,
    text: PropTypes.string,
    isComplete: PropTypes.bool,
    addListItem: PropTypes.func,
    updateItem: PropTypes.func
};