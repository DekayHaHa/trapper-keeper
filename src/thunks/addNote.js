import { hasError, newNote } from '../actions';
// import { getNotes } from './getNotes';

// ADD NOTE
export const addNote = (data) => {
  const url = 'http://localhost:3001/api/notes'
  const option = {
    method: "POST",
    body: JSON.stringify(data),
    headers: {
      "Content-Type": "application/json"
    }
  }
  return async (dispatch) => {
    try {
      const response = await fetch(url, option);
      if(!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      dispatch(newNote(data));
    } catch(error) {
      dispatch(hasError(error.message))
    }
  }
}