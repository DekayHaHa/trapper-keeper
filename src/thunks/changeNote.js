import { hasError, editNote } from '../actions';

export const changeNote = (note) => {
	const option = {
		method: 'PUT',
		body: JSON.stringify(note),
		headers: {
			'Content-Type': 'application/json'
		}
	}
	const url = `http://localhost:3001/api/notes/${note.id}`;
	return async (dispatch) => {
		try {
			const response = await fetch(url, option);
			if (!response.ok) {
				throw Error(response.statusText);
			}
			dispatch(editNote(note));
		} catch (error) {
			dispatch(hasError(error.message));
		}
	}
}