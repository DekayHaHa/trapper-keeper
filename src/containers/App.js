import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Header } from '../components/Header';
import { getNotes } from '../thunks/getNotes';
import NotesContainer from './NotesContainer';
import { Route, Switch } from 'react-router-dom';
import CreateNote from '../containers/CreateNote'
import { PageNotFound } from '../components/PageNotFound';

export class App extends Component {

  componentDidMount() {
    this.props.getNotes();
  }

  findNote = (match) => {
    const { notes } = this.props;
    const { id } = match.params;
    const note = notes.find(note => note.id === id);
    return note
  }

  render() {
    return (
      <div>
        <Route path='/' component={Header} />
        <Switch>
          <Route path='/' exact component={NotesContainer} />
          <Route path='/api/notes/:id' exact render={({ match }) => {
            const note = this.findNote(match)
            return note ? <CreateNote edit={true} {...note} /> : <PageNotFound />
          }} />
          <Route path='/api/new-note' exact component={NotesContainer} />
          <Route component={PageNotFound} />
        </Switch>
      </div>
    );
  }
}

App.propTypes = {
  notes: PropTypes.array,
  getNotes: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
  getNotes: () => dispatch(getNotes())
});

export const mapStateToProps = state => ({
  notes: state.notes
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
