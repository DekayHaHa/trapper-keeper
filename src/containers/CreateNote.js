import React, { Component } from "react";
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addNote } from '../thunks/addNote';
import { ItemTextField } from './ItemTextField';
import { Dialog, DialogTitle, DialogActions, Tooltip, TextField, Button } from '@material-ui/core';
import { changeNote } from '../thunks/changeNote';
import { Redirect } from 'react-router-dom'

export class CreateNote extends Component {
    constructor() {
        super();
        this.state = {
            title: '',
            itemsList: [],
            open: false,
            redirect: false
        };
    }

    componentWillMount() {
        const { title, itemsList } = this.props;
        if (title) {
            this.setState({ title, itemsList, open: true })
        }
    }

    handleChange = (e) => {
        const { value, name } = e.target;
        this.setState({ [name]: value });
    }

    addListItem = (text) => {
        const { itemsList } = this.state;
        if (text) {
            const newItem = { text, isComplete: false, id: Date.now() };
            this.setState({
                itemsList: [...itemsList, newItem],
                item: ''
            });
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        const { title, itemsList } = this.state;
        const { edit, changeNote, addNote, id } = this.props;
        const data = { title, itemsList };
        if (edit) {
            changeNote({ id, title, itemsList });
            this.setState({ itemsList: [], title: '', open: false, redirect: true }, () => {
                this.setState({ redirect: false })
            })
        } else {
            addNote(data);
            this.setState({ itemsList: [], title: '', open: false, redirect: true }, () => {
                this.setState({ redirect: false })
            })
        }
    }

    toggleComplete = (id) => {
        const { itemsList } = this.state;
        const newItems = itemsList.map(item => {
            return id === item.id ? { ...item, isComplete: !item.isComplete } : item
        });
        this.setState({ itemsList: newItems });
    }

    updateItem = (item) => {
        const { itemsList } = this.state;

        const newItems = itemsList.map(val => {
            return item.id === val.id ? { ...item } : val
        });
        this.setState({ itemsList: newItems.filter(val => val.text) });
    }

    checkRedirect = () => {
        const { redirect, open } = this.state
        if (!open && redirect) {
            return <Redirect to='/' />
        }
    }

    renderItems = () => {
        return this.state.itemsList.map((item) => {
            return <ItemTextField key={item.id} {...item} toggle={this.toggleComplete} updateItem={this.updateItem} />;
        });
    };

    handleClickOpen = () => {
        this.setState({ open: true });
    };

    handleClose = () => {
        this.setState({ open: false });
    };

    render() {
        const { open, title } = this.state;

        return (
            <div>
                {this.checkRedirect()}
                <Tooltip title='Create Note' placement='bottom'>
                    <Button color="primary" onClick={this.handleClickOpen}><span className='add-note-btn'>+</span></Button>
                </Tooltip>
                <Dialog open={open} onClose={this.handleClose} aria-labelledby="form-dialog-title" transitionDuration={800} className='dialog-box'>
                    <DialogTitle>
                        <TextField autoFocus margin="dense" id="title" label="Title" type="text" name='title' value={title} onChange={this.handleChange} fullWidth />
                        {
                            this.renderItems()
                        }
                        <ItemTextField addListItem={this.addListItem} />
                    </DialogTitle>
                    <DialogActions>
                        <form>
                            <Button type='submit' onClick={this.handleSubmit} color="primary">Save Note</Button>
                        </form>
                    </DialogActions>
                </Dialog>
            </div>
        );
    }
}
CreateNote.propTypes = {
    id: PropTypes.string,
    title: PropTypes.string,
    itemsList: PropTypes.array,
    edit: PropTypes.bool,
    addNote: PropTypes.func,
    changeNote: PropTypes.func
};

export const mapDispatchToProps = (dispatch) => ({
    addNote: (newNote) => dispatch(addNote(newNote)),
    changeNote: (note) => dispatch(changeNote(note))
});

export default connect(null, mapDispatchToProps)(CreateNote);