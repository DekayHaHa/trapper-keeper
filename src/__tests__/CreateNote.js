import React from 'react';
import { CreateNote, mapDispatchToProps } from '../containers/CreateNote';
import { addNote } from '../thunks/addNote';
import { changeNote } from '../thunks/changeNote';
import { shallow } from 'enzyme';
import { ItemTextField } from '../containers/ItemTextField';
import { Redirect } from 'react-router-dom'

jest.mock('../thunks/addNote');
jest.mock('../thunks/changeNote');

describe('', () => {
	let wrapper;
	let mockFn;
	beforeEach(() => {
		mockFn = jest.fn();
		wrapper = shallow(
			<CreateNote changeNote={mockFn} addNote={mockFn} />
		);
	})

	it('matches snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should have default state', () => {
		expect(wrapper.state()).toEqual({
			title: '',
			itemsList: [],
			open: false,
			redirect: false
		})
	})

	it('should set state if props are passed', () => {
		const mockNote = { title: 'title', itemsList: ['items'] }
		wrapper = shallow(
			<CreateNote {...mockNote} addNote={mockFn} />
		)
		expect(wrapper.state()).toEqual({
			title: 'title',
			itemsList: ['items'],
			open: true,
			redirect: false
		})
	})

	it('should update state depending on event', () => {
		let mockEvent = { target: { name: 'title', value: 'this title' } }
		wrapper.find('#title').simulate('change', mockEvent)
		expect(wrapper.state('title')).toEqual('this title')
	})

	it('should add list item to state', () => {
		wrapper.instance().addListItem('new item');
		expect(wrapper.state('itemsList')[0].text).toEqual('new item')
	})

	describe('handle submit', () => {
		const mockEvent = { preventDefault: () => { } }

		it.skip('should fire changeNote with new edit if edit is true', () => {
			wrapper.instance().handleSubmit(mockEvent)
			expect(wrapper.props.changeNote).toHaveBeenCalled()
		})

		it.skip('should fire addNote with data if edit is false', () => {

		})

		it('should reset state', () => {
			wrapper.setState({
				title: 'title',
				itemsList: ['items'],
				open: true,
				redirect: true
			})
			wrapper.instance().handleSubmit(mockEvent)
			expect(wrapper.state()).toEqual({
				title: '',
				itemsList: [],
				open: false,
				redirect: false
			})
		})
	})

	it('should toggle a single items complete key', () => {
		wrapper.setState({
			itemsList: [
				{ id: 1, isComplete: false },
				{ id: 2, isComplete: false }
			]
		})
		wrapper.instance().handleIsComplete(2)
		expect(wrapper.state('itemsList')).toEqual([
			{ id: 1, isComplete: false },
			{ id: 2, isComplete: true }
		])
	})

	it('should update single item by id', () => {
		wrapper.setState({
			itemsList: [
				{ id: 1, isComplete: false, text: 'first' },
				{ id: 2, isComplete: false, text: 'second' }
			]
		})
		wrapper.instance().updateItem({ id: 2, isComplete: true, text: 'found' })
		expect(wrapper.state('itemsList')).toEqual([
			{ id: 1, isComplete: false, text: 'first' },
			{ id: 2, isComplete: true, text: 'found' }
		])
	})

	it.skip('should set state when handleClickOpen is fired', () => {
		wrapper.find('.add-note-btn').simulate('click');
		expect(wrapper.state('open')).toBe(true);
	});

	it('should set state when handleClickOpen is fired', () => {
		wrapper.find('.dialog-box').simulate('close');
		expect(wrapper.state('open')).toBe(false);
		expect(wrapper.state('redirect')).toBe(true);
	});

	it('should redirect after saving note', () => {
		wrapper.setState({ redirect: true })
		const result = wrapper.instance().checkRedirect()
		expect(result).toEqual(<Redirect to="/" />)
	})

	it('should remove item by id', () => {
		wrapper.setState({
			itemsList: [
				{ id: 1, isComplete: false, text: 'first' },
				{ id: 2, isComplete: false, text: 'second' }
			]
		})
		wrapper.instance().removeItem(1)
		expect(wrapper.state('itemsList')).toEqual([
			{ id: 2, isComplete: false, text: 'second' }
		])
	})

	describe('mapDispatchToProps', () => {
		it('calls dispatch with a addNote action', async () => {
			const mockDispatch = jest.fn();
			const mockItem = { title: "title", itemsList: [] }
			const actionToDispatch = addNote(mockItem);

			const mappedProps = mapDispatchToProps(mockDispatch);
			mappedProps.addNote(mockItem);
			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
		});

		it('calls dispatch with a changeNote action', async () => {
			const mockDispatch = jest.fn();
			const mockNote = { title: "title", itemsList: [] }
			const actionToDispatch = changeNote(mockNote);

			const mappedProps = mapDispatchToProps(mockDispatch);
			mappedProps.changeNote(mockNote);
			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
		});
	});
})