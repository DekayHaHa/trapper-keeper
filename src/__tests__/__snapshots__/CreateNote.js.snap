// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateNote
    addNote={[MockFunction]}
    changeNote={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <WithStyles(Tooltip)
          placement="bottom"
          title="Create Note"
        >
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            <span
              className="add-note-btn"
            >
              +
            </span>
          </WithStyles(Button)>
        </WithStyles(Tooltip)>,
        <WithStyles(Dialog)
          aria-labelledby="form-dialog-title"
          className="dialog-box"
          onClose={[Function]}
          open={false}
          transitionDuration={800}
        >
          <WithStyles(DialogTitle)>
            <TextField
              fullWidth={true}
              id="title"
              label="Title"
              margin="dense"
              name="title"
              onChange={[Function]}
              required={false}
              select={false}
              type="text"
              value=""
              variant="standard"
            />
            <ItemTextField
              addListItem={[Function]}
              addNewInput={[Function]}
              autofocus={true}
              handleIsComplete={[Function]}
              removeItem={[Function]}
            />
          </WithStyles(DialogTitle)>
          <WithStyles(DialogActions)>
            <form
              className="form-btns"
            >
              <WithStyles(Button)
                color="primary"
              >
                +
              </WithStyles(Button)>
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
                type="submit"
              >
                Save Note
              </WithStyles(Button)>
            </form>
          </WithStyles(DialogActions)>
        </WithStyles(Dialog)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            <span
              className="add-note-btn"
            >
              +
            </span>
          </WithStyles(Button)>,
          "placement": "bottom",
          "title": "Create Note",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span
              className="add-note-btn"
            >
              +
            </span>,
            "color": "primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "+",
              "className": "add-note-btn",
            },
            "ref": null,
            "rendered": "+",
            "type": "span",
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-labelledby": "form-dialog-title",
          "children": Array [
            <WithStyles(DialogTitle)>
              <TextField
                fullWidth={true}
                id="title"
                label="Title"
                margin="dense"
                name="title"
                onChange={[Function]}
                required={false}
                select={false}
                type="text"
                value=""
                variant="standard"
              />
              <ItemTextField
                addListItem={[Function]}
                addNewInput={[Function]}
                autofocus={true}
                handleIsComplete={[Function]}
                removeItem={[Function]}
              />
            </WithStyles(DialogTitle)>,
            <WithStyles(DialogActions)>
              <form
                className="form-btns"
              >
                <WithStyles(Button)
                  color="primary"
                >
                  +
                </WithStyles(Button)>
                <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  type="submit"
                >
                  Save Note
                </WithStyles(Button)>
              </form>
            </WithStyles(DialogActions)>,
          ],
          "className": "dialog-box",
          "onClose": [Function],
          "open": false,
          "transitionDuration": 800,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TextField
                  fullWidth={true}
                  id="title"
                  label="Title"
                  margin="dense"
                  name="title"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="text"
                  value=""
                  variant="standard"
                />,
                Array [
                  <ItemTextField
                    addListItem={[Function]}
                    addNewInput={[Function]}
                    autofocus={true}
                    handleIsComplete={[Function]}
                    removeItem={[Function]}
                  />,
                ],
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "fullWidth": true,
                  "id": "title",
                  "label": "Title",
                  "margin": "dense",
                  "name": "title",
                  "onChange": [Function],
                  "required": false,
                  "select": false,
                  "type": "text",
                  "value": "",
                  "variant": "standard",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "1554847576258"
                "nodeType": "class",
                "props": Object {
                  "addListItem": [Function],
                  "addNewInput": [Function],
                  "autofocus": true,
                  "handleIsComplete": [Function],
                  "removeItem": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <form
                className="form-btns"
              >
                <WithStyles(Button)
                  color="primary"
                >
                  +
                </WithStyles(Button)>
                <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  type="submit"
                >
                  Save Note
                </WithStyles(Button)>
              </form>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <WithStyles(Button)
                    color="primary"
                  >
                    +
                  </WithStyles(Button)>,
                  <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    Save Note
                  </WithStyles(Button)>,
                ],
                "className": "form-btns",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "+",
                    "color": "primary",
                  },
                  "ref": null,
                  "rendered": "+",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Save Note",
                    "color": "primary",
                    "onClick": [Function],
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Save Note",
                  "type": [Function],
                },
              ],
              "type": "form",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <WithStyles(Tooltip)
            placement="bottom"
            title="Create Note"
          >
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              <span
                className="add-note-btn"
              >
                +
              </span>
            </WithStyles(Button)>
          </WithStyles(Tooltip)>,
          <WithStyles(Dialog)
            aria-labelledby="form-dialog-title"
            className="dialog-box"
            onClose={[Function]}
            open={false}
            transitionDuration={800}
          >
            <WithStyles(DialogTitle)>
              <TextField
                fullWidth={true}
                id="title"
                label="Title"
                margin="dense"
                name="title"
                onChange={[Function]}
                required={false}
                select={false}
                type="text"
                value=""
                variant="standard"
              />
              <ItemTextField
                addListItem={[Function]}
                addNewInput={[Function]}
                autofocus={true}
                handleIsComplete={[Function]}
                removeItem={[Function]}
              />
            </WithStyles(DialogTitle)>
            <WithStyles(DialogActions)>
              <form
                className="form-btns"
              >
                <WithStyles(Button)
                  color="primary"
                >
                  +
                </WithStyles(Button)>
                <WithStyles(Button)
                  color="primary"
                  onClick={[Function]}
                  type="submit"
                >
                  Save Note
                </WithStyles(Button)>
              </form>
            </WithStyles(DialogActions)>
          </WithStyles(Dialog)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              <span
                className="add-note-btn"
              >
                +
              </span>
            </WithStyles(Button)>,
            "placement": "bottom",
            "title": "Create Note",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span
                className="add-note-btn"
              >
                +
              </span>,
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "+",
                "className": "add-note-btn",
              },
              "ref": null,
              "rendered": "+",
              "type": "span",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-labelledby": "form-dialog-title",
            "children": Array [
              <WithStyles(DialogTitle)>
                <TextField
                  fullWidth={true}
                  id="title"
                  label="Title"
                  margin="dense"
                  name="title"
                  onChange={[Function]}
                  required={false}
                  select={false}
                  type="text"
                  value=""
                  variant="standard"
                />
                <ItemTextField
                  addListItem={[Function]}
                  addNewInput={[Function]}
                  autofocus={true}
                  handleIsComplete={[Function]}
                  removeItem={[Function]}
                />
              </WithStyles(DialogTitle)>,
              <WithStyles(DialogActions)>
                <form
                  className="form-btns"
                >
                  <WithStyles(Button)
                    color="primary"
                  >
                    +
                  </WithStyles(Button)>
                  <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    Save Note
                  </WithStyles(Button)>
                </form>
              </WithStyles(DialogActions)>,
            ],
            "className": "dialog-box",
            "onClose": [Function],
            "open": false,
            "transitionDuration": 800,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <TextField
                    fullWidth={true}
                    id="title"
                    label="Title"
                    margin="dense"
                    name="title"
                    onChange={[Function]}
                    required={false}
                    select={false}
                    type="text"
                    value=""
                    variant="standard"
                  />,
                  Array [
                    <ItemTextField
                      addListItem={[Function]}
                      addNewInput={[Function]}
                      autofocus={true}
                      handleIsComplete={[Function]}
                      removeItem={[Function]}
                    />,
                  ],
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "fullWidth": true,
                    "id": "title",
                    "label": "Title",
                    "margin": "dense",
                    "name": "title",
                    "onChange": [Function],
                    "required": false,
                    "select": false,
                    "type": "text",
                    "value": "",
                    "variant": "standard",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "1554847576258",
                  "nodeType": "class",
                  "props": Object {
                    "addListItem": [Function],
                    "addNewInput": [Function],
                    "autofocus": true,
                    "handleIsComplete": [Function],
                    "removeItem": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <form
                  className="form-btns"
                >
                  <WithStyles(Button)
                    color="primary"
                  >
                    +
                  </WithStyles(Button)>
                  <WithStyles(Button)
                    color="primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    Save Note
                  </WithStyles(Button)>
                </form>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <WithStyles(Button)
                      color="primary"
                    >
                      +
                    </WithStyles(Button)>,
                    <WithStyles(Button)
                      color="primary"
                      onClick={[Function]}
                      type="submit"
                    >
                      Save Note
                    </WithStyles(Button)>,
                  ],
                  "className": "form-btns",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "+",
                      "color": "primary",
                    },
                    "ref": null,
                    "rendered": "+",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Save Note",
                      "color": "primary",
                      "onClick": [Function],
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Save Note",
                    "type": [Function],
                  },
                ],
                "type": "form",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
